docs/schema-changes.md

```diff
@@ -0,0 +1,139 @@
+
+## Data Model
+
+The schema changes data model is composed of the following components:
+
+- `head`: The registry manifest of the most recent version of the semantic
+  convention registry used in the diffing process.
+- `baseline`: The registry manifest of the oldest version of the semantic
+  convention registry used in the diffing process.
+- `changes`: A dictionary of changes between the head and baseline registries.
+  The dictionary is composed of the following keys (when applicable):
+  - `registry_attributes`: A list of changes to registry attributes.
+  - `metrics`: A list of changes to metrics.
+  - `events`: A list of changes to events.
+  - `spans`: A list of changes to spans.
+  - `resources`: A list of changes to resources.
+
+Each change in the changes dictionary for any key is represented as a list of
+schema changes, represented by one of the following change types:
+
+- `added`: A top-level telemetry object (e.g., attribute, metric, etc.) was added to the head registry. The new item’s
+  name is stored in the name attribute.
+- `renamed`: A top-level telemetry object from the baseline registry was renamed in the head registry.
+- `updated`: One or more fields in a top-level telemetry object have been updated in the head registry.
+- `obsoleted`: A top-level telemetry object that is now discontinued without a valid replacement in the head registry.
+- `uncategorized`: A placeholder for complex or unclear schema changes that do not fit into existing types. This type
+  serves as a fallback when no specific category applies, with the expectation that some of these changes will be
+  reclassified into more precise schema types in the future.
+- `removed`: A top-level telemetry object from the baseline registry was removed in the head registry.
+
+> Note: Although the removed schema change type is a valid output of the diffing
+process, it should never be present in the diff report between two versions of
+well-formed semantic convention registries. The policy for semantic convention
+registries is to deprecate items instead of removing them.
+
+Some of these schema change types are based on the new `deprecated` structured format present in the semantic
+conventions. This field is now supporting one of the three variants:
+
+Variant 1
+```yaml
+brief: <text explaining the reason of the renaming>
+deprecated:
+  reason: renamed
+  renamed_to: <name of the telemetry object>
+```
+
+Variant 2
+```yaml
+brief: <text explaining the reason of the obsolescence>
+deprecated:
+  reason: obsoleted
+```
+
+Variant 3
+```yaml
+brief: <text explaining the reason of this complex deprecation>
+deprecated:
+  reason: uncategorized
+```
+
+Example Schema Diff in YAML
+
+```yaml
+head:
+  semconv_version: v1.27.0
+baseline:
+  semconv_version: v1.26.0
+changes:
+  registry_attributes:
+    - name: http.server_name      # attribute name
+      type: obsoleted             # change type
+      note: This attribute is deprecated.
```

Comment:
is the note populated from note or brief?

Reply:
@lmolkova @jerbly Currently, the `note` in the schema change event is directly derived from the `brief` field of the telemetry object. Alternatively, we could choose to use the `note` field of the telemetry object. However, what we cannot do is allow users to fill in either one, as Weaver would then not know which one to use.

One possible solution/extension, however, would be to introduce a `note` field within the `deprecated` field (in the semantic conventions). If this field is populated, it would be used to fill the `note` field in the schema change event. If the `deprecated.note` field is not present, Weaver could then fall back to `brief`.

For this PR, I propose using `brief` for now and introducing the `deprecated.note` field if this approach works for you.